src/chain/electrum.rs

```diff
@@ -0,0 +1,48 @@
+// This file is Copyright its original authors, visible in version control history.
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. You may not use this file except in
+// accordance with one or both of these licenses.
+
+use crate::error::Error;
+use crate::logger::{log_error, LdkLogger, Logger};
+
+use lightning_transaction_sync::ElectrumSyncClient;
+
+use bdk_electrum::BdkElectrumClient;
+
+use electrum_client::Client as ElectrumClient;
+
+use std::sync::Arc;
+
+pub(crate) struct ElectrumRuntimeClient {
+	electrum_client: Arc<ElectrumClient>,
+	bdk_electrum_client: Arc<BdkElectrumClient<ElectrumClient>>,
+	tx_sync: Arc<ElectrumSyncClient<Arc<Logger>>>,
+	runtime: Arc<tokio::runtime::Runtime>,
+	logger: Arc<Logger>,
+}
+
+impl ElectrumRuntimeClient {
```

Comment:
Is something like `tokio::runtime::Handle::current()` useable to still allow init in the builder without a runtime passed in?

Reply:
If it clears up the code, why wait for the refactoring? You'd need to revise the code that is now newly added. And that potential refactoring may never happen especially if it becomes bigger and bigger.

Reply:
Isn't it the case that in the main `start` just a single runtime is created and used everywhere?

Reply:
It's still not entirely clear to me how it could happen that the current handle is different from the one passed into the main start method. But I can see your argument that this is just how to works now.